payroll/models/hr_salary_rule_category.py

```diff
@@ -26,6 +26,15 @@ class HrSalaryRuleCategory(models.Model):
         default=lambda self: self.env.company,
     )
 
+    def _sum_salary_rule_category(self, localdict, amount):
+        self.ensure_one()
+        if self.parent_id:
+            localdict = self.parent_id._sum_salary_rule_category(localdict, amount)
+        localdict["categories"].dict[self.code] = (
+            localdict["categories"].dict.get(self.code, 0) + amount
+        )
+        return localdict
+
```

Comment:
This function always used localdict and amount as params if that is what you are saying. 

The purpose to include it there is because you can need accumulate the categories outside the payslip calculation. 

For example I have a module that makes a report based on accumulated values in categories. If I only can see them in the payslip calculation I have to create a identical method which might be writing again code that is already in the module. 

Also allows to modify this calculation function without the need to rewrite all the get payslip lines function to inject something there. 

My current known about object oriented coding also tells me that this function belong there because it's a method of the category, the category itself offers that functionality. But it think that is more a common dev discussion haha. 

I'm more into of separating concerns in the objects they belong. 

I think it makes no harm to have it there but I could be wrong. What's true is that this function need to be separated from the loop in order to be inherited and modified. 

If you want to modify how categories are calculated now it's easy, before it was rewriting all the function.